haproxy_agent.py

```diff
@@ -251,6 +272,30 @@ async def nc_get_exapp(app_id: str) -> ExApp | None:
         return ExApp.model_validate(data)
 
 
+USER_INFO_URL = f"{ \
```

Comment:
1. let's apply  `removesuffix('/').removesuffix('/index.php')` to the `NC_INSTANCE_URL` on top, instead of applying it to the `USER_INFO_URL` and `EX_APP_URL` separately.
2. typo: `nc_get_user` uses `EX_APP_URL` here and not the `USER_INFO_URL`
3. I suggest to move both `EX_APP_URL`,  `USER_INFO_URL` and `EXCLUDE_HEADERS_USER_INFO` to the top of file

Reply:
> 2. typo: nc_get_user uses EX_APP_URL here and not the USER_INFO_URL

oops